### PR TITLE
[14.0][MIG] connector_prestashop

### PR DESCRIPTION
**Pull request by hailangvn**
This depends on below PR.
 * https://github.com/OCA/server-tools/pull/2083
 * https://github.com/OCA/sale-workflow/pull/1590
 * https://github.com/OCA/product-attribute/pull/857
 * https://github.com/OCA/connector/pull/405
 * https://github.com/OCA/connector-ecommerce/pull/60

### CODE REVIEW

**Review by leemannd — APPROVED** [review]

**Review by StephaneMangin — APPROVED** [review]

**Comment by leemannd** [comment]
Hello @pedrobaeza Do you know which OCA group is able to review and merge PRs on this repo?

**Comment by pedrobaeza** [comment]
This one is on the edge of PSCs, not being something of general Odoo, but I can launch the merge if there are several of you that supports it. I see that you should squash a bit the 14.0 migration commits, and maybe some of the history coming from the same author that are close together in the list.

**Comment by leemannd** [comment]
@pedrobaeza Thanks for the answer. We are using it into a client's project. So if there is need for polishing or fixes, we will handle them.

@hailangvn Could you have a look at P.Baeza 's comment?

**Comment by hailangvn** [comment]
Thanks @leemannd, @StephaneMangin, @pedrobaeza for your support. I have squashed commits, drop test commit. Please let me know if anything else is needed.

**Comment by pedrobaeza** [comment]
OK, thanks, there's still dependencies that need to be merged first.

**Comment by d33x** [comment]
Any idea when this connector will be compatible with Odoo v14? Thanks

File: connector_prestashop/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    "depends": [
+        "account",
+        "base_vat",  # for vat validation on partner address
+        "product",
+        "product_multi_category",  # oca/product-attribute
+        "product_multi_image",  # oca/product-attribute
+        "connector_ecommerce",  # oca/connector-ecommerce
+        "purchase",
+        "onchange_helper",
+        "product_brand",  # oca/product-attribute
```
Comment (florian-dacosta) [inline connector_prestashop/__manifest__.py]: @hailangvn hailangvn
First, thanks for the huge work on the module.
I'll try do do a full review in the future, just a remark for now : 

I believe the connector_prestashop is already really big with a lot of dependencies, we should try not to add anymore.
Indeed, the design is made to make the module quite easily overridable and I think feature likes brand/manufacturer should be in a separate module.
Not every body will use this feature so best to keep in separated to ease further migrations of the base module.

Of course, there are other stuff that could be part of a separated module (refund import for instance ?) but it is in the main module historically.

For instance, the last new feature I needed, the import of product of type "pack", I've made in a separated module (https://github.com/akretion/connector-prestashop/tree/14-connector_prestashop_import_bom)
For the same reason, not every body needs it and it is best for maintenance to keep it separated.

Would you mind putting the brand stuff in an other module ?

**Comment by hailangvn** [comment]
In case `product_brand` is needed, branch https://github.com/hailangvn/connector-prestashop/tree/14.0-imp-connector_prestashop-product_brand can be used.

**Comment by flotho** [comment]
any chance to check ruboat issue?

**Comment by flotho** [comment]
hi @hailangvn , product multi image is about to be merged.
Any chance to review the runboat build issues : 
```
 Couldn't load module connector_prestashop 
2022-04-07 02:14:00,706 232 CRITICAL b56b702da-cc65-46af-b132-c48f6be71599 odoo.modules.module: No module named 'importlib.metadata' 
2022-04-07 02:14:00,707 232 WARNING b56b702da-cc65-46af-b132-c48f6be71599 odoo.modules.loading: Transient module states were reset 
2022-04-07 02:14:00,708 232 ERROR b56b702da-cc65-46af-b132-c48f6be71599 odoo.modules.registry: Failed to load registry

2022-04-07 02:14:00,708 232 CRITICAL b56b702da-cc65-46af-b132-c48f6be71599 odoo.service.server: Failed to initialize database `b56b702da-cc65-46af-b132-c48f6be71599`. 
Traceback (most recent call last):
  File "/opt/odoo/odoo/service/server.py", line 1199, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/opt/odoo/odoo/modules/registry.py", line 89, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/odoo/modules/loading.py", line 461, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/opt/odoo/odoo/modules/loading.py", line 349, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/odoo/odoo/modules/loading.py", line 179, in load_module_graph
    load_openerp_module(package.name)
  File "/opt/odoo/odoo/modules/module.py", line 385, in load_openerp_module
    __import__('odoo.addons.' + module_name)
  File "/mnt/data/odoo-addons-dir/connector_prestashop/__init__.py", line 5, in <module>
    from . import components
  File "/mnt/data/odoo-addons-dir/connector_prestashop/components/__init__.py", line 5, in <module>
    from . import backend_adapter
  File "/mnt/data/odoo-addons-dir/connector_prestashop/components/backend_adapter.py", line 7, in <module>
    from prestapyt import PrestaShopWebServiceDict, PrestaShopWebServiceError
  File "/opt/odoo-venv/lib/python3.6/site-packages/prestapyt/__init__.py", line 1, in <module>
    from .prestapyt import PrestaShopWebService
  File "/opt/odoo-venv/lib/python3.6/site-packages/prestapyt/prestapyt.py", line 52, in <module>
    from .version import __author__
  File "/opt/odoo-venv/lib/python3.6/site-packages/prestapyt/version.py", line 1, in <module>
    from importlib.metadata import version, PackageNotFoundError
ModuleNotFoundError: No module named 'importlib.metadata'
```

**Comment by hailangvn** [comment]
Good day @flotho, the `prestapyt` package needs python >= 3.8 and travis can be set in [this commit](https://github.com/OCA/connector-prestashop/pull/147/commits/c254cbcae6be5fa09df4ee8c6d2fc9b8732d255d). But runboat is using `/opt/odoo-venv/lib/python3.6` (as you can see in the error log above) and I not yet know how to change it.
Good day @sbidoul, could you please advise? Thanks in advance.

**Comment by flotho** [comment]
Hi @rousseldenis @gurneyalex @sbidoul @simahawk 

Any tips to make runboat work with this, this awesome PR has never been closer to a merge !

**Comment by sbidoul** [comment]
As far as I can tell the prestapyt project has been updated to support python 3.6, but has not made a release for that yet.
Asking the prestapyt maintainers to cut a release should resolve the issue.

**Comment by pedrobaeza** [comment]
I have done the release: https://github.com/prestapyt/prestapyt/releases/tag/0.9.2, but not sure if the PyPi thing is done automatically. I'm only maintaining the library as it's orphaned, but we are not using it since several years.

**Comment by sbidoul** [comment]
@pedrobaeza apparently it gets published to PyPI automatically: 
image

But you should rather do a 0.11.0

**Comment by pedrobaeza** [comment]
Let's see now: https://github.com/prestapyt/prestapyt/releases/tag/0.11.0

The version history seems a mess

**Comment by hailangvn** [comment]
Thanks @pedrobaeza, @sbidoul. The runboat is green now.

**Comment by pedrobaeza** [comment]
Why is green if one dependency is not migrated yet?

**Comment by flotho** [comment]
> Why is green if one dependency is not migrated yet?

good question, what part of the process is concerned ? ocaBot ?

**Comment by sbidoul** [comment]
> Why is green if one dependency is not migrated yet?

Because there is a git reference to the dependency migration PR in `test-requirements.txt`. That is fine.

**Comment by pedrobaeza** [comment]
Ah, ok, is when finally merging when it turns red. Please remove such git reference and we will be able to merge.

**Comment by flotho** [comment]
hi  @hailangvn , you're almost done on this one, last easy request from @pedrobaeza 
Tell us if it's clear

**Comment by hailangvn** [comment]
Thanks @pedrobaeza , @flotho. The tmp commit has been removed.

**Comment by flotho** [comment]
hi @pedrobaeza is it normal that your merge tentative hasn't succeeded ?

**Comment by flotho** [comment]
Hi @pedrobaeza , sadlytravis failed on  
```
BlockingIOError: [Errno 11] write could not complete without blocking
```
How can I help on this ?

**Comment by pedrobaeza** [comment]
Please rebase that I have switched to GH actions

**Comment by flotho** [comment]
hi @hailangvn , any chance to have an additionnal rebase on this one ?

**Comment by hailangvn** [comment]
Thanks @pedrobaeza, @flotho. This has been rebased.

**Pull request merged**